### PR TITLE
Add preconditions to consumer for refundAddress and fundingKey

### PR DESCRIPTION
**Pull request by braydonf**

### CODE REVIEW

**Comment by pnagurny** [comment]
LGTM

File: lib/consumer.js [hunk 1]
```diff
@@ -71,7 +71,8 @@ function Consumer(opts) {
    * @desc A private key for funding the channel. An alternative implementation could
    * provide a list of unspent outputs and the keys needed to sign the outputs
    */
-  this.fundingKey = opts.fundingKey || new PrivateKey();
+  $.checkArgument(opts.fundingKey instanceof PrivateKey, 'fundingKey is expected to be a PrivateKey');
+  this.fundingKey = opts.fundingKey;
```
Comment (maraoz) [inline lib/consumer.js]: This changes the behavior. Previously, if no fundingKey was supplied, it would create a new one for you.

Comment (braydonf) [inline lib/consumer.js]: Is there a use case for that? Seems doing such would make it easy to lose the private key.

Comment (maraoz) [inline lib/consumer.js]: Hmm you may be right. thoughts @eordano ?

**Comment by eordano** [comment]
ACK

**Comment by aynik** [comment]
So I think now the guide is wrong?

http://bitcore.io/guide/module/channel/index.html

**Pull request merged**